packages/eslint-plugin/src/rules/no-unused-vars.ts

```diff
@@ -255,6 +255,17 @@ export default util.createRule<Options, MessageIds>({
     }
 
     return {
+      // children of a namespace that is exported in a definition file is autoExported
+      [ambientDeclarationSelector(
+        'TSModuleDeclaration > TSModuleBlock',
+        false,
+      )](node: DeclarationSelectorNode): void {
```

Comment:
I think this is too far-reaching.
This selector will in essence overwrite the selector on line 288.
Additionally this selector will apply to any and every module declaration, and not just top-level ones.

A better solution would be to augment the selector on line 270 so that it handles exports.

```
'Program, Program > ExportNamedDeclaration'
```

Reply:
I assumed that since it only applies for definition files having a new selector would be fine. Will make the change on 270 itself.